dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/impl/DynamoSettings.scala

```diff
@@ -5,20 +5,59 @@
 package akka.stream.alpakka.dynamodb.impl
 
 import akka.actor.ActorSystem
+import com.amazonaws.auth._
+import com.typesafe.config.Config
+
+import scala.util.Try
 
 object DynamoSettings {
-  def apply(system: ActorSystem): DynamoSettings = {
-    val config = system.settings.config.getConfig("akka.stream.alpakka.dynamodb")
+
+  /**
+   * Scala API: Creates [[DynamoSettings]] from the [[com.typesafe.config.Config Config]] attached to an [[akka.actor.ActorSystem]].
+   */
+  def apply(system: ActorSystem): DynamoSettings = apply(system.settings.config)
+
+  /**
+   * Scala API: Creates [[DynamoSettings]] from a [[com.typesafe.config.Config Config]]
+   */
+  def apply(baseConfig: Config): DynamoSettings = {
+    val config = baseConfig.getConfig("akka.stream.alpakka.dynamodb")
+    val awsCredentialsProvider = Try(config.getConfig("credentials"))
```

Comment:
Please test existence with `hasPath` instead.

Reply:
Is there a specific reason for this? `Try().getOrElse()` does the same thing and `hasPath` forces an `if`. I'm quite happy to, just curious.

Reply:
Switch to using `hasPath`

Reply:
Doing this with `hasPath` also breaks the fallback to default credentials is the config is not fully defined, i.e. missing the `credentials.secret-key-id` which is handled tidily with the `Try` form.
If you want me to check both paths explicitly then I can do so, but it seems much more long winded to do so.